src/main/java/ui/components/pickers/PickerMilestone.java

```diff
@@ -63,6 +59,56 @@ public Node getNewlyAssignedMilestoneNode() {
         return milestone;
     }
 
+    public Node getDetailedMilestoneNode() {
+        HBox matchingMilestone = new HBox();
+        matchingMilestone.setSpacing(3);
+        matchingMilestone.setPadding(new Insets(3, 3, 3, 3));
+        matchingMilestone.setStyle("-fx-border-width: 0 0 1 0; -fx-border-style: solid;");
+
+        HBox milestoneNodeBox = createMilestoneNodeBox();
+        Label separator = new Label("|");
+        HBox milestoneDetailsBox = createMilestoneDetailsBox();
+
+        matchingMilestone.getChildren().setAll(milestoneNodeBox, separator, milestoneDetailsBox);
+        return matchingMilestone;
+    }
+
+    private HBox createMilestoneDetailsBox() {
+        HBox milestoneDetailsBox = new HBox();
+        milestoneDetailsBox.setSpacing(3);
+        milestoneDetailsBox.setPrefWidth(250);
+        milestoneDetailsBox.setAlignment(Pos.CENTER_RIGHT);
+
+        if (getDueDate().isPresent()) {
+            Label dueDate = new Label("Due on: " + getDueDate().get().toString());
+            dueDate.setPrefWidth(150);
+            milestoneDetailsBox.getChildren().add(dueDate);
+        }
+        int totalIssues = getOpenIssues() + getClosedIssues();
+        double progressValue = totalIssues > 0
+                ? (double) getClosedIssues() / (getOpenIssues() + getClosedIssues()) : 0;
```

Comment:
Now that this is being done in `PickerMilestone`, I think we can extract getting `progressValue` into a separate method and compute it in the constructor instead. We can store its result in a class variable. This is so that it does not need to re-compute `progressValue` every time this is called.

Also, do change the `(getOpenIssues() + getClosedIssues())` to `totalIssues` which I missed out the last time. :stuck_out_tongue: